mkraft/node/asnoleader.go

```diff
@@ -166,7 +170,14 @@ func (n *Node) RunAsCandidate(ctx context.Context) {
 				case <-electionTicker.C:
 					// last election reaches no decisive result, either no response or not enough votes
 					// we re-elect
-					consensusChan = n.candidateAsyncDoElection(ctx)
+					electionCancel()
+					electionCtx, electionCancel = context.WithCancel(ctx)
+					defer electionCancel()
```

Comment:
defer in a loop is not very practical as it may create a long defer func stack. If we want to setup this kind of logic we may do this:
```go
// before `for`, like in the code
electionCtx, electionCancel := context.WithTimeout(ctx, timeout)
// Notice the wrapping in func.
defer func() { electionCancel() }

for {
        // ...
        electionCancel()
        electionCtx, electionCancel = WithTimeout(ctx, timeout)
        // no defer here
}
```